File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/ChatCollectionView/ChatOutgoingCell/ChatOutgoingCell.m
//
//  ChatOutgoingCell.m
//  sample-chat
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "ChatOutgoingCell.h"
#import "UIView+Chat.h"
#import "CALayer+Chat.h"

@implementation ChatOutgoingCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    UIColor *bubbleColor = [UIColor colorWithRed:0.22f green:0.47f blue:0.99 alpha:1.0f];
    self.previewContainer.backgroundColor = bubbleColor;
    [self.layer applyShadowWithColor:bubbleColor alpha:0.4f forX:0.0f forY:12.0f blur:12.0f spread:0.0f path:nil];
}

+ (ChatCellLayoutModel)layoutModel {
    
    ChatCellLayoutModel defaultLayoutModel = [super layoutModel];
    defaultLayoutModel.avatarSize = CGSizeMake(0, 0);
    defaultLayoutModel.containerInsets = UIEdgeInsetsMake(0, 16, 12, 14);
    defaultLayoutModel.spaceBetweenTopLabelAndTextView = 12.0f;
    defaultLayoutModel.timeLabelHeight = 15.0f;
    
    return defaultLayoutModel;
}

- (void)layoutSubviews {
    CGFloat cornerRadius = 20.0f;
    CAShapeLayer *layer = CAShapeLayer.new;
    layer.frame = self.previewContainer.layer.bounds;
    UIRectCorner roundingCorners = UIRectCornerBottomLeft | UIRectCornerTopLeft | UIRectCornerTopRight;
    UIBezierPath *bPath = [UIBezierPath bezierPathWithRoundedRect:self.previewContainer.bounds
                                                byRoundingCorners:roundingCorners
                                                      cornerRadii:CGSizeMake(cornerRadius, cornerRadius)];
    layer.path = bPath.CGPath;
    self.previewContainer.layer.mask = layer;
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/Chat/CollectionView/CollectionCells/ChatOutgoingCell/ChatOutgoingCell.swift
//
//  ChatOutgoingCell.swift
//  Swift-ChatViewController
//
//  Created by Injoit on 1/28/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class ChatOutgoingCell: ChatCell {
    
    @IBOutlet weak var statusImageView: UIImageView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        previewContainer.backgroundColor = #colorLiteral(red: 0.2218334079, green: 0.4693790674, blue: 0.9888214469, alpha: 1)
        layer.applyShadow(color: #colorLiteral(red: 0.2216441333, green: 0.4713830948, blue: 0.9869660735, alpha: 1), alpha: 0.4, y: 12.0, blur: 12.0)
    }
    
    override class func layoutModel() -> ChatCellLayoutModel {
        let containerInsets = UIEdgeInsets(top: 0.0, left: 16.0, bottom: 12.0, right: 16.0)
        var defaultLayoutModel = super.layoutModel()
        defaultLayoutModel.avatarSize = .zero
        defaultLayoutModel.containerInsets = containerInsets
        defaultLayoutModel.timeLabelHeight = 15.0
        defaultLayoutModel.spaceBetweenTopLabelAndTextView = 12.0
        
        return defaultLayoutModel
    }
    
    override func layoutSubviews() {
        let roundingCorners:UIRectCorner = [.bottomLeft, .topLeft, .topRight]
        let layer = CAShapeLayer()
        layer.frame = previewContainer.layer.bounds
        let bPath = UIBezierPath(roundedRect: previewContainer.bounds,
                                 byRoundingCorners: roundingCorners,
                                 cornerRadii: CGSize(width: 20, height: 20))
        layer.path = bPath.cgPath
        previewContainer.layer.mask = layer
    }
    
    func setupStatusImage(_ image: UIImage) {
        statusImageView.image = image
    }
}
